File: CS50X/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_valid_int(void);
void create_pyramid(int n);

int main(void)
{
    int height = get_valid_int();
    create_pyramid(height);
}

void create_pyramid(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int x = n; x > i + 1; x--)
        {
            printf(" ");
        }
        for (int x = 1; x <= i + 1; x++)
        {
            printf("#");
        }
        printf("  ");
        for (int x = 1; x <= i + 1; x++)
        {
            printf("#");
        }
        printf("\n");
    }
}

int get_valid_int(void)
{
    while (true)
    {
        int valid_int = get_int("Height: ");
        if (valid_int > 0)
        {
            return valid_int;
        }
    }
}

File: CS50X/sentimental-mario-more/mario.py
def main():
    height = get_int()
    create_pyramid(height)

def get_int():
    while True:
        usernumber = input("Height: ")
        try:
            if int(usernumber) > 0 and int(usernumber) < 9:
                return int(usernumber)
        except ValueError:
            continue

def create_pyramid(height):
    for x in range(height):
        a = height
        while a > x + 1:
            print(" ",end="")
            a -= 1
        b = 1
        while b <= x + 1:
            print("#",end="")
            b += 1
        print("  ",end="")
        c = 1
        while c <= x + 1:
            print("#",end="")
            c += 1
        print("\n",end="")

if __name__ == "__main__":
    main()
